lib/fastlane/plugin/wpmreleasetoolkit/helper/android/android_localize_helper.rb

```diff
@@ -299,6 +299,7 @@ def self.merge_xml_documents(all_xmls)
         #
         def self.apply_substitutions(tag)
           tag.content = tag.content.gsub('...', '…')
+          tag.content = tag.content.gsub('-', '&#8211;')
```

Comment:
TIL.

How come this substitution then becomes `&amp;#8211;` in the generated strings?

Reply:
@JavonDavis Wait, who added this change? I remember explicitly explaining in another PR why this was a bad idea 😓  because it will solve some linter errors but introduce others (en dash should only be used instead of standard dash in certain circumstances – especially, for ranges – but not all, so replacing all instances is semantically and typographically incorrect, and will introduce the bug for the other use case)

We discussed in 229 that this kind of misuse of the dash (use of regular dash when en dash is expected for a range, use of en dash when regular dash is expected for hyphenation and other cases, …) had to be solved in GlotPress directly, either by the translators or by ourselves (it's an easy enough change that we don't need to know the language to edit the GlotPress entry), but definitively not automatically and unconditionally for all cases.

Reply:
Because we're replacing `content`, which is an XML text node. It's not up to us to do the serialization/encoding, it's what the XML lib (Nokogiri) handles for us. So by setting `tag.content` to a value with `&#8211;` we are telling Nokogiri that this is the literal text that we want. Then Nokogiri does its job that ensures that the text content is properly escaped (encode `<` as `&lt;`, `>` as `&gt;`, `&` as `&amp;`, …) when it serializes the nodes as XML document.